### PR TITLE
Azure Monitor Exporter - Modified ExtractRoleInfo to read roleinstance and rolename once 

### PR DESCRIPTION
**Pull request by rajkumar-rangaraj**
* Introduced two internal properties `RoleName` and `RoleInstance` to avoid repeatedly reading role name and instance.

### CODE REVIEW

File: sdk/monitor/Microsoft.Azure.Monitor.OpenTelemetry.Exporter/src/AzureMonitorConverter.cs [hunk 1]
```diff
@@ -71,18 +75,22 @@ internal static TelemetryItem GeneratePartAEnvelope(Activity activity)
             return telemetryItem;
         }
 
-        internal static void ExtractRoleInfo(Resource resource, out string roleName, out string roleInstance)
```
Comment (TimothyMothra) [inline sdk/monitor/Microsoft.Azure.Monitor.OpenTelemetry.Exporter/src/AzureMonitorConverter.cs:74]: nit: consider renaming this to something like `InitRoleInfo`.
In my opinion, that's more representative of what this method is doing. 
But totally fine to leave it as it is. Your call. :)

Comment (rajkumar-rangaraj) [reply sdk/monitor/Microsoft.Azure.Monitor.OpenTelemetry.Exporter/src/AzureMonitorConverter.cs:74]: I was looking out for a good name for this method. `InitRoleInfo` looks good to me.

File: sdk/monitor/Microsoft.Azure.Monitor.OpenTelemetry.Exporter/tests/Microsoft.Azure.Monitor.OpenTelemetry.Exporter.Tests/AzureMonitorConverterTests.cs [hunk 2]
```diff
@@ -30,68 +30,81 @@ static AzureMonitorConverterTests()
...
 
         [Fact]
         public void ExtractRoleInfo_ServiceInstance()
         {
-            var resource = Resources.CreateServiceResource(null, "roleInstance_1");
-            AzureMonitorConverter.ExtractRoleInfo(resource, out var roleName, out var roleInstance);
-            Assert.Empty(resource.Attributes);
-            Assert.Null(roleName);
-            Assert.Null(roleInstance);
+            using var activity = new Activity("ExtractRoleInfo_ServiceName");
```
Comment (TimothyMothra) [inline sdk/monitor/Microsoft.Azure.Monitor.OpenTelemetry.Exporter/tests/Microsoft.Azure.Monitor.OpenTelemetry.Exporter.Tests/AzureMonitorConverterTests.cs:72]: nit: i don't think this matters, but in your above tests the Activity name matches the test name. these last tests don't.

**Review by TimothyMothra — APPROVED** [review]
LGTM

File: sdk/monitor/Microsoft.Azure.Monitor.OpenTelemetry.Exporter/tests/Microsoft.Azure.Monitor.OpenTelemetry.Exporter.Tests/AzureMonitorConverterTests.cs (hunk 2 above)
Comment (rajkumar-rangaraj) [reply sdk/monitor/Microsoft.Azure.Monitor.OpenTelemetry.Exporter/tests/Microsoft.Azure.Monitor.OpenTelemetry.Exporter.Tests/AzureMonitorConverterTests.cs]: Fixed.

**Pull request merged**